presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestApproximatePercentileAggregation.java

```diff
@@ -320,125 +361,206 @@ public void testFloatPartialStep()
+        testAggregationDoubles(
                 DOUBLE_APPROXIMATE_PERCENTILE_WEIGHTED_AGGREGATION,
-                1.0,
-                createDoublesBlock(null, 1.0),
-                createLongsBlock(1L, 1L),
-                createRLEBlock(0.5, 2));
+                new Page(doublesBlock, weightsBlock, percentiles),
+                error,
+                ImmutableList.of(quantile),
+                inputs);
+        // Test with weights and accuracy
+        testAggregationDoubles(
+                DOUBLE_APPROXIMATE_PERCENTILE_WEIGHTED_AGGREGATION_WITH_PARAMETER,
+                new Page(doublesBlock, weightsBlock, percentiles, BlockAssertions.createRLEBlock(parameter, doublesBlock.getPositionCount())),
+                error,
+                ImmutableList.of(quantile),
+                inputs);
+    }
 
-        assertAggregation(
-                DOUBLE_APPROXIMATE_PERCENTILE_WEIGHTED_AGGREGATION,
-                2.0,
-                createDoublesBlock(null, 1.0, 2.0, 3.0),
-                createLongsBlock(1L, 1L, 1L, 1L),
-                createRLEBlock(0.5, 4));
+    private void testArrayAggregationDouble(Block doublesBlock, Block weightsBlock, Block percentiles, double error, List<Double> quantiles, double... inputs)
+    {
+        // Test without weights and without accuracy/compression
+        testAggregationDoubles(
+                DOUBLE_APPROXIMATE_PERCENTILE_ARRAY_AGGREGATION,
+                new Page(doublesBlock, percentiles),
+                error,
+                quantiles,
+                inputs);
+        // Test with weights and without accuracy/compression
+        testAggregationDoubles(
+                DOUBLE_APPROXIMATE_PERCENTILE_ARRAY_WEIGHTED_AGGREGATION,
+                new Page(doublesBlock, weightsBlock, percentiles),
+                error,
+                quantiles,
+                inputs);
+    }
 
-        assertAggregation(
-                DOUBLE_APPROXIMATE_PERCENTILE_WEIGHTED_AGGREGATION,
-                2.0,
-                createDoublesBlock(1.0, 2.0, 3.0),
-                createLongsBlock(1L, 1L, 1L),
-                createRLEBlock(0.5, 3));
+    private void testAggregationDoubles(InternalAggregationFunction function, Page page, double error, List<Double> quantiles, double... inputs)
+    {
+        // test scalars
+        List<Double> rows = Arrays.stream(inputs).sorted().boxed().collect(Collectors.toList());
+        List<Double> returned;
+        try {
+            Double value = (Double) AggregationTestUtils.aggregation(function, page);
+            if (value == null) {
+                returned = new ArrayList<>();
+            }
+            else {
+                returned = ImmutableList.of(value);
+            }
+        }
+        catch (ClassCastException e) {
+            returned = (List<Double>) AggregationTestUtils.aggregation(function, page);
+        }
+        assertPercentilesWithinError(returned, error, rows, quantiles);
```

Comment:
Let's keep the order of the parameters consistent.  If `quantiles` comes first above, let's keep it in the same order below.